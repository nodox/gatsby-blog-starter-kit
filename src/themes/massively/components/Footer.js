import React from 'react';

export default class Footer extends React.Component {

  render() {
    return (
      <footer id="footer">
        <section>
          <form method="post"
                name="contact"
                action="/thanks/"
                data-netlify="true"
                data-netlify-honeypot="bot-field"
                encType="application/x-www-form-urlencoded">
            <input type="hidden" name="form-name" value="contact" />
            <p hidden>
             <label>
               Donâ€™t fill this out: <input name="bot-field" />
             </label>
            </p>
            <div className="field">
              <label>Name</label>
              <input type="text" name="name" id="name" placeholder="Your name"/>
            </div>
            <div className="field">
              <label>Email</label>
              <input id="email" type="email" name="email" placeholder="Your email"/>
            </div>
            <div className="field">
              <label>Message</label>
              <textarea id="message" name="message" rows="3" placeholder="Your message"></textarea>
            </div>
            <ul className="actions">
              <li><input type="submit" value="Send Message"/></li>
            </ul>
          </form>
        </section>
        <section className="split contact">
          <section>
            <h3>Social</h3>
            <ul className="icons alt">
              <li><a href="https://www.twitter.com/stevennatera" className="icon alt fa-twitter"><span className="label">Twitter</span></a></li>
              <li><a href="https://www.linkedin.com/in/snatera" className="icon alt fa-linkedin"><span className="label">Facebook</span></a></li>
              <li><a href="https://www.github.com/nodox" className="alt icon fa-github"><span className="label">GitHub</span></a></li>
            </ul>
          </section>
        </section>
      </footer>
    );
  }
}







