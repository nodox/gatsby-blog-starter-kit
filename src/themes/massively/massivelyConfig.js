module.exports = {
    initialLinks: {
      paths: [
        {
          path: '/',
          title: 'Home',
        },
        {
          path: '/about',
          title: 'About'
        },
        {
          path: '/speaking',
          title: 'Speaking'
        },
        {
          path: '/contact',
          title: 'Contact'
        },
      ],
      socialLinks: [
        {
          path: 'https://www.twitter.com/stevennatera',
          icon: 'fa-twitter',
          name: 'Twitter',
        },
        {
          path: 'https://www.linkedin.com/in/snatera',
          icon: 'fa-linkedin',
          name: 'LinkedIn',
        },
        {
          path: 'https://www.github.com/nodox',
          icon: 'fa-github',
          name: 'Github',
        }
      ]
    },
    copyright: {
      owner: 'Â© Steven Natera',
    },
    introText: {
      '/': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/about': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/speaking': 'I deliver talks on NodeJS and I teach software engineering workshops.',
      '/contact': 'I\'m pretty easy to get hold of, here\'s how I use different channels to communicate with people and how best to contact me'
    }
}
