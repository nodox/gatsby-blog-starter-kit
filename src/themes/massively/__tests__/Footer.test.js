import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../components/Footer';
import * as config from '../massivelyConfig';

describe('<Footer />', () => {
  const component = mount(
    <Footer navLinks={config.initialLinks}/>
  );
  const props = component.instance().props;

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('navLinks');
    expect(props).toHaveProperty('navLinks.socialLinks');
  });

});
