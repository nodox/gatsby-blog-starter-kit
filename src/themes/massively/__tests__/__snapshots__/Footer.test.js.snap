// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<Footer
  config={
    Object {
      "copyright": Object {
        "owner": "© Steven Natera",
      },
      "initialLinks": Object {
        "paths": Array [
          Object {
            "path": "/",
            "title": "Home",
          },
          Object {
            "path": "/about",
            "title": "About",
          },
          Object {
            "path": "/speaking",
            "title": "Speaking",
          },
          Object {
            "path": "/contact",
            "title": "Contact",
          },
        ],
        "socialLinks": Array [
          Object {
            "icon": "fa-twitter",
            "name": "Twitter",
            "path": "https://www.twitter.com/stevennatera",
          },
          Object {
            "icon": "fa-linkedin",
            "name": "LinkedIn",
            "path": "https://www.linkedin.com/in/snatera",
          },
          Object {
            "icon": "fa-github",
            "name": "Github",
            "path": "https://www.github.com/nodox",
          },
        ],
      },
      "introText": Object {
        "/": "Hi! I'm Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.",
        "/about": "Hi! I'm Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.",
        "/contact": "I'm pretty easy to get hold of, here's how I use different channels to communicate with people and how best to contact me",
        "/speaking": "I deliver talks on NodeJS and I teach software engineering workshops.",
      },
      "name": "Steven Natera",
    }
  }
>
  <footer
    id="footer"
  >
    <section>
      <form
        action="/thanks/"
        data-netlify="true"
        data-netlify-honeypot="bot-field"
        encType="application/x-www-form-urlencoded"
        method="post"
        name="contact"
      >
        <input
          name="form-name"
          type="hidden"
          value="contact"
        />
        <p
          hidden={true}
        >
          <label>
            Don’t fill this out: 
            <input
              name="bot-field"
            />
          </label>
        </p>
        <div
          className="field"
        >
          <label>
            Name
          </label>
          <input
            id="name"
            name="name"
            placeholder="Your name"
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Email
          </label>
          <input
            id="email"
            name="email"
            placeholder="Your email"
            type="email"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Message
          </label>
          <textarea
            id="message"
            name="message"
            placeholder="Your message"
            rows="3"
          />
        </div>
        <ul
          className="actions"
        >
          <li>
            <input
              type="submit"
              value="Send Message"
            />
          </li>
        </ul>
      </form>
    </section>
    <section
      className="split contact"
    >
      <section>
        <h3>
          Social
        </h3>
        <ul
          className="icons alt"
        >
          <li
            key="0"
          >
            <a
              className="icon fa-twitter alt"
              href="https://www.twitter.com/stevennatera"
            >
              <span
                className="label"
              >
                Twitter
              </span>
            </a>
          </li>
          <li
            key="1"
          >
            <a
              className="icon fa-linkedin alt"
              href="https://www.linkedin.com/in/snatera"
            >
              <span
                className="label"
              >
                LinkedIn
              </span>
            </a>
          </li>
          <li
            key="2"
          >
            <a
              className="icon fa-github alt"
              href="https://www.github.com/nodox"
            >
              <span
                className="label"
              >
                Github
              </span>
            </a>
          </li>
        </ul>
      </section>
    </section>
  </footer>
</Footer>
`;
